### ISSUE TITLE
Alert user to fill its profile account and channels

### ISSUE THREAD

**Issue by kimsible** [Chocobozzz/PeerTube#3331]
I would be nice to find a way to inform users to put an avatar and description in their profile empty account and channels.

Would you think it's a good idea or really too intrusive ?

I think about 2 diferrents ways : 

- A discrete alert in My account or My library below the sub-menu ;
- A modal like the admin modal to help the user to configure its profile account and channels.

**Comment by test2a** [comment]
Should this be in first login when a user is most likely to add one? 
Something like how websites after signup say "one more thing before you get started... here. Update your profile" where a user would add display name, description, photo, support "

There would be a skip option obviously but this should be discussed.

**Comment by kimsible** [comment]
Thanks for your comment !

 > Should this be in first login when a user is most likely to add one?

Every first login and when we create a new channel without uploading an avatar and fill a description.

> There would be a skip option obviously but this should be discussed.

This indeed should have a skip option. 

> Something like how websites after signup say "one more thing before you get started... here. Update your profile" where a user would add display name, description, photo, support "

Yes it could be very good but, I may be wrong, it should be discreet but visible, what I don't like with modal is that it should only be used for a very important warning (like important rule or announcement) but not to help a UX problem.

**Comment by test2a** [comment]
Another idea. At signup, we have this step 1, 2 3. How about in the same vein a step 4 after registration and right after email verification, to configure these things ?

**Comment by kimsible** [comment]
Yes it seems a pretty good idea, we need to make sure that this 4th step is also displayed for already registered / created accounts

**Comment by rigelk** [comment]
We could copy the current modal on first login that is used for administrators to configure their server.

**Comment by kimsible** [comment]
Thanks ! it sounds that is the quickest solution technically and is consistent with the current tools, @test2a what do you think about this proposition ?

**Comment by test2a** [comment]
Fine by me I guess. Lets have something up and then we can improve upon it.

**Comment by test2a** [comment]
second thing, when we create a new channel, 
should we also add the option to upload image because that way the whole thing would be done in a single page
![Screenshot_20201120_144633](https://user-images.githubusercontent.com/39449563/99782535-4bd33900-2b3f-11eb-8805-98ca96257116.png)


the same way like we have in "edit" channel mode.

![a](https://user-images.githubusercontent.com/39449563/99782717-89d05d00-2b3f-11eb-917c-7bd019a64162.png)

**Comment by kimsible** [comment]
True @test2a , the update method of the avatar component should be handled with the button form at creation for this.


What do you think about editing avatar on the public view ?
![image](https://user-images.githubusercontent.com/1877318/99889447-abbc0380-2c55-11eb-8e57-be3cdb692a42.png)

**Comment by rigelk** [comment]
@kimsible isn't that a bit far from solving the initial problem?

**Comment by kimsible** [comment]
You're right @rigelk but I thought maybe if we need to alert the user, it is that there is a conception issue with editing avatar, isn't it ?

**Comment by rigelk** [comment]
I see where you are coming from, but the issue already identified the problem lies with alerting and instructing the user, not transforming a display view into an editing view. I get your intent to make things easier, but we are just one click away from doing just that with the "manage account" button anyway.

**Comment by kimsible** [comment]
@test2a your remark about uploading avatar while channel creation has already it's own issue : https://github.com/Chocobozzz/PeerTube/issues/1506

**Comment by test2a** [comment]
@kimsible seems like downstream is faster. Ha ha. 
Tubelab already has that feature now. 

https://github.com/stom79/TubeLab/commit/1e45e4454c10795da883350d5ccc71992faee7b5
